tests/ui/contracts.rs

```diff
@@ -0,0 +1,23 @@
+//@ revisions: no_fails catch_fail no_checking
+//@ [no_fails]    run-pass
+//@ [catch_fail]  run-fail
+//@ [no_checking] run-pass
+
+//@[no_fails]    compile-flags: -Zcontract-checking=dynamic
+//@[catch_fail]  compile-flags: -Zcontract-checking=dynamic
+//@[no_checking] compile-flags: -Zcontract-checking=none
+
+#![allow(internal_features)]
+#![feature(rustc_attrs, rustc_contracts, core_intrinsics)]
+
+#[contracts::requires(x > 0)]
```

Comment:
I do like `contracts::`, but it just occurred to me that it could conflict with the crate [contracts](https://crates.io/crates/contracts).